### PR TITLE
PLATFORM-2531 log User::crypt calls on production

### PR DESCRIPTION
**Pull request by themech**
Similarly to https://github.com/Wikia/app/pull/11647, I want to track down all the places where passwords are still generated in PHP. Later on, based on the logs, I'll change those places to use Helios.

### CODE REVIEW

File: includes/User.php [hunk 1]
```diff
@@ -4191,7 +4191,18 @@ public static function oldCrypt( $password, $userId ) {
 	 */
 	public static function crypt( $password, $salt = false ) {
 		global $wgPasswordSalt;
-
+		// Wikia change - begin
+		// @see PLATFORM-2502 comparing new passwords in PHP code.
+		// @todo mech remove after the new password hashing is implemented (PLATFORM-2530).
```
Comment (WikiaTech) [inline includes/User.php:4196]: INFO Complete the task associated to this "TODO" comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 134 issues
- CRITICAL 4 critical
- MAJOR 51 major
- MINOR 42 minor
- INFO 37 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [User.php#L961](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L961): Remove this hard-coded password. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
2. CRITICAL [User.php#L2194](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L2194): Remove this hard-coded password. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
3. CRITICAL [User.php#L2198](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L2198): Remove this hard-coded password. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
4. CRITICAL [User.php#L3230](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L3230): Remove this hard-coded password. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
5. MAJOR [User.php](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php): File "User.php" has 4889 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
6. MAJOR [User.php#L70](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L70): Class "User" has 224 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
7. MAJOR [User.php#L125](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L125): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
8. MAJOR [User.php#L131](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L131): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
9. MAJOR [User.php#L136](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L136): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
10. MAJOR [User.php#L143](https://github.com/Wikia/app/blob/239c8a0ac167ae3962fb4be92dd216ebd39d2f19/includes/User.php#L143): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)

**Review by Grunny — APPROVED** [review]
LGTM.

**Comment by ludwikkazmierczak** [comment]
LGTM

**Pull request merged**